lib/ascii_binder/helpers.rb

```diff
@@ -296,10 +297,33 @@ def linkify_breadcrumb(href, text, extra_level)
     end
 
     def parse_distros distros_string, for_validation=false
-      values = distros_string.split(',').map{ |v| v.strip }
-      return values if for_validation
-      return distro_map.keys if values.include?('all')
-      return values.uniq
+      values   = distros_string.split(',').map{ |v| v.strip }
+      expanded = expand_distro_globs(values)
+      return expanded if for_validation
+      return distro_map.keys if expanded.include?('all')
+      return expanded.uniq
+    end
+
+    def expand_distro_globs(values)
+      complete = []
+      values.each do |k|
+        if k == '*'
+          return distro_map.keys
+        elsif k.include?('*')
+          key_parts = k.split('*')
+          if key_parts.length > 2
+            raise "The Distro key value '#{k}' contains too many wildcards; only one is supported."
+          end
+          distro_map.keys.each do |dkey|
+            if (key_parts[0].nil? or dkey.start_with?(key_parts[0])) and (key_parts[1].nil? or dkey.end_with?(key_parts[1]))
+              complete << dkey
+            end
+          end
+        else
+          complete << k
+        end
+      end
```

Comment:
This feels like it would be a lot simpler to just convert to a normal Regex and then apply it (and then it can support multiple wildcards or whatever they want).

```ruby
def expand_distro_globs(values)
  values.flat_map do |value|
    value_regex = Regexp.new("\\A#{value.gsub("*", ".*")}\\z")
    distro_map.keys.select { |k| value_regex.match(k) }
  end.uniq
end

# distro_map = {:community => nil, :community_special => nil, :not_the_community => nil, :maybe_community_but_not_sure => nil}
# expand_distro_globs(["community"])
#  => [:community]
# expand_distro_globs(["community*"]) 
#  => [:community, :community_special]
# expand_distro_globs(["*community"])
#  => [:community, :not_the_community]
# expand_distro_globs(["*community*"])
#  => [:community, :community_special, :not_the_community, :maybe_community_but_not_sure]
# expand_distro_globs(["enterprise"])
#  => []
```